llama-index-integrations/readers/llama-index-readers-oracleai/llama_index/readers/oracleai/base.py

```diff
@@ -0,0 +1,457 @@
+                                    Document(text=str(row[1]), metadata=metadata)
+                                )
+                except Exception as ex:
+                    print(f"An exception occurred :: {ex}")
+                    traceback.print_exc()
+                    cursor.close()
+                    raise
+
+            return results
+        except Exception as ex:
+            print(f"An exception occurred :: {ex}")
+            traceback.print_exc()
+            raise
+
+
+import logging
+from typing import (
+    Any,
+    Dict,
+    List,
+)
+
+
+logger = logging.getLogger(__name__)
+
+
+class OracleTextSplitter:
+    """Splitting text using Oracle chunker."""
+
+    def __init__(self, conn: Connection, params: Dict[str, Any]):
+        self.conn = conn
+        self.params = params
+
+        try:
+            import oracledb
+        except ImportError as e:
+            raise ImportError(
+                "Unable to import oracledb, please install with "
+                "`pip install -U oracledb`."
+            ) from e
+
+        self._oracledb = oracledb
+        self._json = json
+
+    def split_text(self, text: str) -> List[str]:
+        """Split incoming text and return chunks."""
+        splits = []
+
+        try:
+            cursor = self.conn.cursor()
+            # returns strings or bytes instead of a locator
+            self._oracledb.defaults.fetch_lobs = False
+
+            cursor.setinputsizes(content=self._oracledb.CLOB)
+            cursor.execute(
+                "select t.* from dbms_vector_chain.utl_to_chunks(:content, json(:params)) t",
+                content=text,
+                params=self._json.dumps(self.params),
+            )
+
+            while True:
+                row = cursor.fetchone()
+                if row is None:
+                    break
+                d = self._json.loads(row[0])
+                splits.append(d["chunk_data"])
+
+            return splits
+
+        except Exception as ex:
+            print(f"An exception occurred :: {ex}")
+            traceback.print_exc()
+            raise
+
+
+# uncomment the following code block to run the test
+
+"""
+# A sample unit test.
+
+# get the Oracle connection
+conn = oracledb.connect(
+    user="<username>",
+    password="<password>",
+    dsn="<hostname/service_name>",
+)
+print("Oracle connection is established...")
+
+# params
+loader_params = {"owner": "ut", "tablename": "demo_tab", "colname": "data"}
+splitter_params = {"by": "words", "max": "100"}
+
+# instances
+loader = OracleReader(conn=conn, params=loader_params)
+splitter = OracleTextSplitter(conn=conn, params=splitter_params)
+
+print("Processing the documents...")
+docs = loader.load()
+for id, doc in enumerate(docs, start=1):
+    print(f"Document#{id}, Metadata: {doc.metadata}")
+    chunks = splitter.split_text(doc.text)
+    print(f"Document#{id}, Num of Chunk: {len(chunks)}\n")
+
+conn.close()
+print("Connection is closed.")
+
+"""
```

Comment:
This should also be in an actual test file

Reply:
As mentioned before, this is kept handy to do a quick test and verification. We do have actual test file as well. Please let me know if you still want to be removed.

Reply:
removed.